libs/cli/kubeagi_cli/serve/reranking.py

```diff
@@ -15,28 +15,76 @@
 
 from typing import List
 from FlagEmbedding import FlagReranker
+from BCEmbedding import RerankerModel
+from abc import ABC, abstractmethod
 
 
-class Reranking:
+class BaseReranking(ABC):
     """
-    The Reranking is used to run reranking models like bge-reranker-large(https://huggingface.co/BAAI/bge-reranker-large)
+    The Reranking is used to run reranking models.
+    """
+
+    @abstractmethod
+    def run(self, pairs: List[List[str]]):
+        """run reranking models to rerank pairs."""
+
+
+class BGEReranking(BaseReranking):
+    """
+    The BGEReranking is used to run reranking models with FlagEmbedding like sbge-reranker-large(https://huggingface.co/BAAI/bge-reranker-large)
     """
 
-    _model_path: str
     _reranker: FlagReranker
 
     def __init__(
         self,
-        model_path: str,
+        model_name_or_path: str,
+    ):
+        self._reranker = FlagReranker(
+            model_name_or_path=model_name_or_path, use_fp16=False
+        )
+
+    # run bge reranking model
+    def run(self, pairs: List[List[str]]):
+        if len(pairs) > 0:
+            result = self._reranker.compute_score(pairs)
+            if isinstance(result, float):
+                result = [result]
+            return result
+        else:
+            return None
+
+
+class BCEReranking(BaseReranking):
+    """
+    The BGEReranking is used to run reranking models with BCEEmbedding from https://github.com/netease-youdao/BCEmbedding
+    """
+
+    _reranker: RerankerModel
+
+    def __init__(
+        self,
+        model_name_or_path: str,
     ):
-        self._model_path = model_path
+        self._reranker = RerankerModel(
+            model_name_or_path=model_name_or_path, use_fp16=False
+        )
 
+    # run bge reranking model
     def run(self, pairs: List[List[str]]):
-        reranker = FlagReranker(self._model_path, use_fp16=False)
         if len(pairs) > 0:
-            result = reranker.compute_score(pairs)
+            result = self._reranker.compute_score(pairs)
             if isinstance(result, float):
                 result = [result]
             return result
         else:
             return None
+
+
+def select_reranking(model_name_or_path: str) -> BaseReranking:
```

Comment:
look like if we use transformers, we can have same code to serve diffrent rerank model?
```python
import torch
from transformers import AutoTokenizer, AutoModelForSequenceClassification

# init model and tokenizer
tokenizer = AutoTokenizer.from_pretrained('maidalun1020/bce-reranker-base_v1')
model = AutoModelForSequenceClassification.from_pretrained('maidalun1020/bce-reranker-base_v1')

device = 'cuda'  # if no GPU, set "cpu"
model.to(device)

# get inputs
inputs = tokenizer(sentence_pairs, padding=True, truncation=True, max_length=512, return_tensors="pt")
inputs_on_device = {k: v.to(device) for k, v in inputs.items()}

# calculate scores
scores = model(**inputs_on_device, return_dict=True).logits.view(-1,).float()
scores = torch.sigmoid(scores)
```
```python
import torch
from transformers import AutoModelForSequenceClassification, AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained('BAAI/bge-reranker-large')
model = AutoModelForSequenceClassification.from_pretrained('BAAI/bge-reranker-large')
model.eval()

pairs = [['what is panda?', 'hi'], ['what is panda?', 'The giant panda (Ailuropoda melanoleuca), sometimes called a panda bear or simply panda, is a bear species endemic to China.']]
with torch.no_grad():
    inputs = tokenizer(pairs, padding=True, truncation=True, return_tensors='pt', max_length=512)
    scores = model(**inputs, return_dict=True).logits.view(-1, ).float()
    print(scores)
```

Reply:
FlagEmbedding/BCEEmbedding provides interfaces to load reranking models which uses the transformers as well. By using `FlagEmbedding/BCEEmbedding`,we do not need to care much about the base dependecies.